### ISSUE TITLE
C tokenizer exited with non-empty token stack.

### ISSUE THREAD

**Issue by kjschiroo** [earwig/mwparserfromhell#158]
I get the following message when parsing revision id 732147567

```
File "/home/kjschiroo/Development/venvs/3/lib/python3.5/site-packages/mwparserfromhell-0.4.3-py3.5-linux-x86_64.egg/mwparserfromhell/utils.py", line 58, in parse_anything
    return Parser().parse(value, context, skip_style_tags)
  File "/home/kjschiroo/Development/venvs/3/lib/python3.5/site-packages/mwparserfromhell-0.4.3-py3.5-linux-x86_64.egg/mwparserfromhell/parser/__init__.py", line 93, in parse
    tokens = self._tokenizer.tokenize(text, context, skip_style_tags)
mwparserfromhell.parser.ParserError: This is a bug and should be reported. Info: C tokenizer exited with non-empty token stack.
```

Here is the revision text
[732147567.txt](https://github.com/earwig/mwparserfromhell/files/414206/732147567.txt)

**Comment by lahwaacz** [comment]
Hmm, the `skip_style_tags=True` workaround for #40 does not help here.

**Comment by earwig** [comment]
Fixed in cd4f90e663fa421b836e93ddc56e4a573eefb664. Same bug as #147.
